PWG/EMCAL/EMCALbase/AliAnalysisTaskEmcalEmbeddingHelper.cxx

```diff
@@ -615,6 +619,79 @@ bool AliAnalysisTaskEmcalEmbeddingHelper::IsRunInRunlist(const std::string & pat
   return false;
 }
 
+/**
+ * Get the data background run number from the data file path, and determine which run run block of 
+ * fEmbeddedRunblock it belongs to. Then filter the input list of MC files and select those anchored to this run block.
+ * Note that this functionality should only be used when train runs per run.
+ */
+void AliAnalysisTaskEmcalEmbeddingHelper::FilterRunblockFilenames() 
+{
+  std::size_t nBlocks = fEmbeddedRunblock.size();
+ 
+  if ( nBlocks == 0 ) {
+    return ;
+  }
+  
+  std::cout << "AliAnalysisTaskEmcalEmbeddingHelper::FilterRunblockFilenames() - nRunBlocks " << nBlocks << std::endl;
+  
+  // Recover the backgroun data run number
+  //
+  Int_t runNumber = -1;
+  if ( AliAnalysisTaskSE::InputEvent() ) {
+    runNumber = AliAnalysisTaskSE::InputEvent()->GetRunNumber();
+    
+    if ( runNumber < 100000 || runNumber > 300000 ) {
+      AliFatal(Form("Data run number %d not good!",runNumber));
+    }
+  }
+  
+  if ( fDataRunNumber != runNumber ) {
+    AliDebugStream(0) << "    Set data run number, new "<< runNumber << ", previous "<<fDataRunNumber<<"\n";
+    fDataRunNumber = runNumber;
+  }
+  
+  std::cout << "    Data run anchor "<<fDataRunNumber<<"\n";
+  
+  // Select the run block
+  //
+  Int_t runMin = -1, runMax = -1;
+  Int_t iblock = 0;
+  for (iblock = 0; iblock < nBlocks-1; iblock++) 
+  {  
+    runMin = fEmbeddedRunblock.at(iblock  );
+    runMax = fEmbeddedRunblock.at(iblock+1);
+    
+    AliDebug(0,Form("\t block %d, run min %d, run max %d",iblock,runMin,runMax));
+    
+    if ( runMin <= fDataRunNumber && runMax > fDataRunNumber ) {
+      break ;
+    }
+  } // block loop
+  
+  std::cout << "    Selected run range block "<< iblock <<": ["<<runMin<<","<<runMax<<"]"<<std::endl;
+  
+  if ( runMin < 0 && runMax < 0 ) {
+    AliFatal("Runblock not found, stop!");
+    return ;
+  }
+
+  // Filter the list of files
+  //
+  std::cout << "    Size of filenames before filtering "<<  fFilenames.size() << std::endl; 
```

Comment:
When I suggested streams, I was thinking along the lines of `AliDebugStream`, `AliInfoStream`, etc. It would probably be best to cleanup these `std::cout`

Reply:
At some point I was not seeing the prints and then used cout, now changed to AliInfoStream and AliDebugStream